### ISSUE TITLE
Add support for inventory slot filters

### ISSUE THREAD

**Issue by LevFendi** [Factorio-Access/FactorioAccess#190]
Vanilla supports this for player inventory and cargo wagon slots. 

The basic way to do this is to press an keybond on a slot that has an item in it and set the slot filter into that item.

A good candidate for the keybind is ALT + LEFTBRACKET.

**Comment by EphDoering** [comment]
I think we should support all three modes of selection plus clearing:
1. press when slot is full to filter on that item
2. press when slot is empty with item in hand to filter on the item in hand
3. press when slot is empty and no item in hand to enter the select item menu (refactor to be able to re-use filter selecting menu)
4. press when slot is already filtered to clear the filter.

Also, we should just check if the inventory we're in supports filters since mods can set weather something is filterable.

**Comment by LevFendi** [comment]
1, 2, 4, and the supporting check are easy to do, so I will do those first. I feel that 3 is gated behind the upcoming UI overhaul.

**Comment by ahicks92** [comment]
Yeah 3 is very blocked on uI overhaul.  Which is my priority after KK and I probably get 2-3 weeks off soon.

**Comment by LevFendi** [comment]
All except 3 have been added to main now.
